### PR TITLE
feat(DDM): add ddm instances data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ddm instances data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ddm instances data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ddm/' TESTARGS='-run TestAccDatasourceDdmInstances_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ddm/ -v -run TestAccDatasourceDdmInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDdmInstances_basic
=== PAUSE TestAccDatasourceDdmInstances_basic
=== CONT  TestAccDatasourceDdmInstances_basic
--- PASS: TestAccDatasourceDdmInstances_basic (454.70s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ddm       454.750s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/ddm/data_source_huaweicloud_ddm_instances_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func testAccDatasourceDdmInstances_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_ddm_instances" "test" {
+  depends_on = [huaweicloud_ddm_instance.test]
+}
+`, testAccDatasourceDdmInstances_base(name))
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ddm/data_source_huaweicloud_ddm_instances_test.go]: I think here could reuse `testDdmInstance_basic()` in `resource_huaweicloud_ddm_instance_test.go`.
The above function `testAccDatasourceDdmInstances_base` coule be removed.

Comment (houpeng80) [reply huaweicloud/services/acceptance/ddm/data_source_huaweicloud_ddm_instances_test.go]: done

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_instances.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceDdmInstances() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceDdmInstancesRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_instances.go:29]: I think we shoud support more query options for the data source, for example: name, status, enterprise_project_id

File: docs/data-sources/ddm_instances.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  engine_id         = data.huaweicloud_ddm_engines.test.engines[0].id
+  vpc_id            = huaweicloud_vpc.test.id
+  subnet_id         = huaweicloud_vpc_subnet.test.id
+  security_group_id = huaweicloud_networking_secgroup.test.id
+
+  availability_zones = [
+    data.huaweicloud_availability_zones.test.names[0]
+  ]
+}
+
```
Comment (ShiChangkuo) [inline docs/data-sources/ddm_instances.md:13]: the above codes can be deleted, you can use variables to instead them

Comment (houpeng80) [reply docs/data-sources/ddm_instances.md:13]: done

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_instances.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/data_source_huaweicloud_ddm_instances.go:29]: done

**Pull request merged**